spec/checkMergeConflictSpec.js

```diff
@@ -0,0 +1,261 @@
+require('dotenv').config();
+const { createProbot } = require('probot');
+// The plugin refers to the actual app in index.js.
+const oppiaBot = require('../index');
+const checkMergeConflictModule = require('../lib/checkMergeConflicts');
+const scheduler = require('../lib/scheduler');
+let payloadData = JSON.parse(
+  JSON.stringify(require('../fixtures/pullRequestPayload.json'))
+);
+
+describe('Merge Conflict Check', () => {
+  /**
+   * @type {import('probot').Probot} robot
+   */
+  let robot;
+
+  /**
+   * @type {import('probot').Octokit} github
+   */
+  let github;
+
+  /**
+   * @type {import('probot').Application} app
+   */
+  let app;
+
+  let mergeConflictLabel = {
+    id: 971968901,
+    node_id: 'MDU6TGFiZWw5NzE5Njg5MDE=',
+    url:
+      "https://api.github.com/repos/oppia/oppia/labels/PR:%20don't%20merge%20-%20HAS%20MERGE%20CONFLICTS",
+    name: "PR: don't merge - HAS MERGE CONFLICTS",
+    color: 'd93f0b',
+    default: false,
+  };
+  beforeEach(() => {
+    spyOn(scheduler, 'createScheduler').and.callFake(() => {});
+
+    github = {
+      issues: {
+        addLabels: jasmine.createSpy('addLabels').and.resolveTo({}),
+        addAssignees: jasmine.createSpy('addAssignees').and.resolveTo({}),
+        createComment: jasmine.createSpy('createComment').and.resolveTo({}),
+        removeLabel: jasmine.createSpy('removeLabel').and.resolveTo({}),
+      },
+    };
+
+    robot = createProbot({
+      id: 1,
+      cert: 'test',
+      githubToken: 'test',
+    });
+
+    app = robot.load(oppiaBot);
+    spyOn(app, 'auth').and.resolveTo(github);
+    spyOn(
+      checkMergeConflictModule,
+      'checkMergeConflictsInAllPullRequests'
+    ).and.callThrough();
+    spyOn(
+      checkMergeConflictModule,
+      'checkMergeConflictsInPullRequest'
+    ).and.callThrough();
+  });
+
+  describe('pull request merge results in merge conflict', () => {
+    beforeEach(async () => {
+      // Simulate a merged pull request.
+      payloadData.payload.pull_request.merged = true;
+      payloadData.payload.action = 'closed';
+
+      const pullRequest = { ...payloadData.payload.pull_request };
```

Comment:
I am a bit confused regarding this -- are we simulating a merge on a PR first and then merge conflict on the same PR?

Reply:
So what I did here is to use 1 PR as if it was 2 PRs, which is why I'm doing the object spread.
The pull request from the payload is the one that triggers the event, while the new PR is what will be gotten when the API is called and the second one needs to to be set as not merged and not mergeable (merge conflict).

Reply:
> Is there any issue with doing that?

Not really, but there will be a lot of repetition of what is in the payload already.